plugins/modules/s3_object_info.py

```diff
@@ -0,0 +1,614 @@
+
+    if len(object_retention_info) != 0 and 'msg' not in object_retention_info.keys():
+        # Remove ResponseMetadata from object_retention_info, convert to snake_case
+        del(object_retention_info['ResponseMetadata'])
+        object_retention_info = camel_dict_to_snake_dict(object_retention_info)
+
+    return object_retention_info
+
+
+def describe_s3_object_tagging(connection, module, bucket_name, object_name):
+    params = {}
+    params['Bucket'] = bucket_name
+    params['Key'] = object_name
+
+    object_tagging_info = {}
+
+    try:
+        object_tagging_info = connection.get_object_tagging(**params)
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        object_tagging_info['msg'] = 'Object tags not found, are you sure the bucket/object tags exist?'
+
+    if len(object_tagging_info) != 0 and 'msg' not in object_tagging_info.keys():
+        # Remove ResponseMetadata from object_tagging_info, convert to snake_case
+        del(object_tagging_info['ResponseMetadata'])
+        object_tagging_info = boto3_tag_list_to_ansible_dict(object_tagging_info['TagSet'])
+
+    return object_tagging_info
+
+
+def get_object_details(connection, module, bucket_name, object_name, requested_facts):
+
+    all_facts = {}
+
+    # Remove non-requested facts
+    requested_facts = {fact: value for fact, value in requested_facts.items() if value is True}
+
+    all_facts['object_metadata'] = get_object(connection, module, bucket_name, object_name)['object_metadata']
+
+    for key in requested_facts:
+        if key == 'object_acl':
+            all_facts[key] = {}
+            all_facts[key] = describe_s3_object_acl(connection, module, bucket_name, object_name)
+        elif key == 'object_attributes':
+            all_facts[key] = {}
+            all_facts[key] = describe_s3_object_attributes(connection, module, bucket_name, object_name)
+        elif key == 'object_legal_hold':
+            all_facts[key] = {}
+            all_facts[key] = describe_s3_object_legal_hold(connection, module, bucket_name, object_name)
+        elif key == 'object_lock_configuration':
+            all_facts[key] = {}
+            all_facts[key] = describe_s3_object_lock_configuration(connection, module, bucket_name)
+        elif key == 'object_retention':
+            all_facts[key] = {}
+            all_facts[key] = describe_s3_object_retention(connection, module, bucket_name, object_name)
+        elif key == 'object_tagging':
+            all_facts[key] = {}
+            all_facts[key] = describe_s3_object_tagging(connection, module, bucket_name, object_name)
+
+    return all_facts
+
+
+def get_object(connection, module, bucket_name, object_name):
+    params = {}
+    params['Bucket'] = bucket_name
+    params['Key'] = object_name
+
+    result = {}
+
+    try:
+        object_info = connection.get_object(**params)
+    except (botocore.exceptions.ClientError, botocore.exceptions.BotoCoreError) as e:
+        pass
```

Comment:
I guess we should fail if some error occurs. We should also handle the case when the specified object is not found.

Reply:
@mandar242 What happened is the specified object does not exist? What if the S3 bucket does not exist?

Reply:
Fixed. Added check(s) to determine if bucket or a object exists and/or has access permissions.